js/modules/k6/html/html.go

```diff
@@ -71,3 +264,191 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
 	}
 	return s.rt.ToValue(val)
 }
+
+func (s Selection) Html() goja.Value {
+	val, err := s.sel.Html()
+	if err != nil {
+		return goja.Undefined()
+	}
+	return s.rt.ToValue(val)
+}
+
+func (s Selection) Val() goja.Value {
+	switch goquery.NodeName(s.sel) {
+	case "input":
+		return s.Attr("value")
+
+	case "textarea":
+		return s.Html()
+
+	case ButtonTagName:
+		return s.Attr("value")
+
+	case "option":
+		return s.rt.ToValue(valueOrHTML(s.sel))
+
+	case "select":
+		selected := s.sel.First().Find("option[selected]")
+
+		if _, exists := s.sel.Attr("multiple"); exists {
+			return s.rt.ToValue(selected.Map(func(idx int, opt *goquery.Selection) string { return valueOrHTML(opt) }))
+		} else {
+			return s.rt.ToValue(valueOrHTML(selected))
+		}
+
+	default:
+		return goja.Undefined()
+	}
+}
+
+func (s Selection) Children(def ...string) Selection {
+	if len(def) == 0 {
+		return Selection{s.rt, s.sel.Children(), s.URL}
+	}
+
+	return Selection{s.rt, s.sel.ChildrenFiltered(def[0]), s.URL}
+}
+
+func (s Selection) Each(v goja.Value) Selection {
+	gojaFn, isFn := goja.AssertFunction(v)
+	if !isFn {
+		panic(s.rt.NewGoError(errors.New("Argument to each() must be a function")))
```

Comment:
Instead of these panic calls, you can use `common.Throw(rt, err)`. That's a helper that makes sure not to double-wrap errors.